pkgs/applications/audio/indicator-sound-switcher/default.nix

```diff
@@ -0,0 +1,65 @@
+{ python3Packages
+, lib
+, fetchurl
+, perlPackages
+, gettext
+, gtk3
+, gobject-introspection
+, intltool, wrapGAppsHook, glib
+, librsvg
+, libayatana-appindicator-gtk3
+, libpulseaudio
+, keybinder3
+, gdk-pixbuf
+}:
+
+let
+  name = "indicator-sound-switcher";
+  version = "2.3.6";
+in
+python3Packages.buildPythonApplication {
+#python3Packages.buildPythonPackage {
+  pname = name;
+  inherit version;
+  src = fetchurl {
+    url = "https://github.com/yktoo/${name}/archive/v${version}.tar.gz";
+    sha256 = "sha256:0xrywmppzrr4x6hhlsgrhbh516djzb04ms717vfbpycy7316arvr";
+  };
+
+  nativeBuildInputs = [
+    gettext
+    intltool
+    wrapGAppsHook
+    glib
+    gdk-pixbuf
+  ];
+
+  buildInputs = [
+    librsvg
+  ];
+
+  propagatedBuildInputs = [
+    python3Packages.setuptools
+    python3Packages.pygobject3
+    gtk3
+    gobject-introspection
+    librsvg
+    libayatana-appindicator-gtk3
+    libpulseaudio
+    keybinder3
+  ];
+
+  makeWrapperArgs = [
+    "--prefix LD_LIBRARY_PATH : ${libpulseaudio}/lib"
```

Comment:
Is this really required? Could we patch the path instead?

Reply:
What does this mean? :)
This works, and I am unsure how to do it in another way.

Reply:
I think it may be possible to instead of letting the python library reference to an arbitrary libpulse.so.0, instead explicitly reference to `${libpulseaudio}/lib/libpulse.so`.

This avoids LD_LIBRARY_PATH to unintentionally propagate to other processes if indicator-sound-switcher executes other programs.

Possibly byI looked at the project and it might be possible to 'patch' lines like:

https://github.com/yktoo/indicator-sound-switcher/blob/e0a159b349580da21809fe6b9b0a221c47317bf4/lib/indicator_sound_switcher/lib_pulseaudio.py#L7

The CDLL call loads the library, so that argument could be changed.

Patching could be done using `substituteInPlace` in `postPatch`.

Reply:
@bobvanderlinden Thanks a lot, I understand it much better and I have made the change now :smile: